### ISSUE TITLE
Support secured tunnel

### ISSUE THREAD

**Issue by AlkindiX** [rpclib/rpclib#181]
rpclib is a great library with awesome features. But it doesn't support secured layer communication between client and the server and this is very important. Implementing Openssl is a great deal and will secure the connection

**Comment by sztomi** [comment]
Yes, this is definitely a shortcoming and will be covered as part of #59 . I'm keeping this open because I don't think there is a separate issue for this now.

**Comment by fhajji** [comment]
OpenSSL is too heavy a dependency for this. Maybe libsodium would be enough?
